diff -r ac61bf8c09a6 hiit.py
--- a/hiit.py	Fri Oct 05 17:50:58 2012 -0700
+++ b/hiit.py	Sun Oct 07 16:21:26 2012 -0700
@@ -1,15 +1,8 @@
-import kivy
-kivy.require('1.4.1')
+from kivy.clock import Clock
+from kivy.properties import ObjectProperty
 
-from kivy.app import App
-from kivy.clock import Clock
-from kivy.config import Config
-from kivy.core.audio import SoundLoader
-from kivy.properties import ObjectProperty
-from kivy.uix.boxlayout import BoxLayout
-from kivy.uix.button import Button
-from kivy.uix.label import Label
-from kivy.uix.popup import Popup
+from timer import GenTimer
+from sequence import Sequence
 
 class HiitTimer(GenTimer):
 
@@ -22,8 +15,8 @@
 
     def __init__(self, **kwargs):
         super(HiitTimer, self).__init__(**kwargs)
-        self.sequence_begin = [self.tick] * 5 + [self.run_timer, self.stop_sequence]
-        self.sequence_end   = [self.tick] * 5 + [self.gong, self.stop_sequence]
+        #self.sequence_begin = [self.clap] * 5 + [self.run_timer, #self.stop_sequence]
+        #self.sequence_end   = [self.clap] * 5 + [self.gong, self.stop_sequence]
 
     def increment_value(self, dt):
         self.progress.value += dt / self.total_duration
@@ -42,7 +35,7 @@
             self.start_sequence(self.sequence_begin)
         else:
             Clock.unschedule(self.increment_value)
-            Clock.unschedule(self.tick)
+            Clock.unschedule(self.clap)
             self.stop_sequence()
             if auto:
                 self.start_sequence(self.sequence_end)
@@ -50,15 +43,11 @@
     def run_timer(self, dt=0):
         Clock.schedule_interval(self.increment_value, .1)
         if self.interval.value:
-            Clock.schedule_interval(self.tick, self.interval.value * 60)
+            Clock.schedule_interval(self.clap, self.interval.value * 60)
         self.gong()
 
-def setdefaults(config):
-    config.setdefaults(config_name, {
-        'num_cycles': '10',
-        'high_intensity': '60',
-        'low_intensity': '90'
-    })
+    def help(self):
+        GenTimer.help(help_title, help_text)
 
 help_title = 'How to Practice High-Intensity Interval Training'
 
diff -r ac61bf8c09a6 main.py
--- a/main.py	Fri Oct 05 17:50:58 2012 -0700
+++ b/main.py	Sun Oct 07 16:21:26 2012 -0700
@@ -6,8 +6,9 @@
 Config.set('graphics', 'height', '800')
 
 from kivy.app import App
-from kivy.core.audio import SoundLoader
 from kivy.factory import Factory
+from kivy.uix.accordion import Accordion
+
 import meditation
 import relaxation
 import yoga
diff -r ac61bf8c09a6 sequence.py
--- a/sequence.py	Fri Oct 05 17:50:58 2012 -0700
+++ b/sequence.py	Sun Oct 07 16:21:26 2012 -0700
@@ -1,6 +1,3 @@
-import kivy
-kivy.require('1.4.1')
-
 from kivy.clock import Clock
 
 class Sequence:
@@ -9,14 +6,20 @@
         self.l = l
         self.interval = i
 
-    def start(self):
+    def start(self, debug=False):
         self.todo = self.l[:]
+        self.debug = debug
         Clock.schedule_interval(self.run_item, self.interval)
-        self.run_sequence()
+        self.run_item(0)
 
     def stop(self):
-        Clock.unschedule(self.run_sequence)
+        Clock.unschedule(self.run_item)
 
     def run_item(self, dt):
-        self.todo.pop(0)(dt)
+        if self.debug:
+            print "(DEBUG  ) (Sequence    )", [ x.__name__ for x in self.todo]
+        if self.todo:
+            self.todo.pop(0)(dt)
+        else:
+          self.stop()
 
