diff -r ac61bf8c09a6 hiit.py
--- a/hiit.py	Fri Oct 05 17:50:58 2012 -0700
+++ b/hiit.py	Sun Oct 07 16:21:26 2012 -0700
@@ -1,15 +1,8 @@
-import kivy
-kivy.require('1.4.1')
+from kivy.clock import Clock
+from kivy.properties import ObjectProperty
 
-from kivy.app import App
-from kivy.clock import Clock
-from kivy.config import Config
-from kivy.core.audio import SoundLoader
-from kivy.properties import ObjectProperty
-from kivy.uix.boxlayout import BoxLayout
-from kivy.uix.button import Button
-from kivy.uix.label import Label
-from kivy.uix.popup import Popup
+from timer import GenTimer
+from sequence import Sequence
 
 class HiitTimer(GenTimer):
 
@@ -22,8 +15,8 @@
 
     def __init__(self, **kwargs):
         super(HiitTimer, self).__init__(**kwargs)
-        self.sequence_begin = [self.tick] * 5 + [self.run_timer, self.stop_sequence]
-        self.sequence_end   = [self.tick] * 5 + [self.gong, self.stop_sequence]
+        #self.sequence_begin = [self.clap] * 5 + [self.run_timer, #self.stop_sequence]
+        #self.sequence_end   = [self.clap] * 5 + [self.gong, self.stop_sequence]
 
     def increment_value(self, dt):
         self.progress.value += dt / self.total_duration
@@ -42,7 +35,7 @@
             self.start_sequence(self.sequence_begin)
         else:
             Clock.unschedule(self.increment_value)
-            Clock.unschedule(self.tick)
+            Clock.unschedule(self.clap)
             self.stop_sequence()
             if auto:
                 self.start_sequence(self.sequence_end)
@@ -50,15 +43,11 @@
     def run_timer(self, dt=0):
         Clock.schedule_interval(self.increment_value, .1)
         if self.interval.value:
-            Clock.schedule_interval(self.tick, self.interval.value * 60)
+            Clock.schedule_interval(self.clap, self.interval.value * 60)
         self.gong()
 
-def setdefaults(config):
-    config.setdefaults(config_name, {
-        'num_cycles': '10',
-        'high_intensity': '60',
-        'low_intensity': '90'
-    })
+    def help(self):
+        GenTimer.help(help_title, help_text)
 
 help_title = 'How to Practice High-Intensity Interval Training'
 
diff -r ac61bf8c09a6 main.py
--- a/main.py	Fri Oct 05 17:50:58 2012 -0700
+++ b/main.py	Sun Oct 07 16:21:26 2012 -0700
@@ -6,8 +6,9 @@
 Config.set('graphics', 'height', '800')
 
 from kivy.app import App
-from kivy.core.audio import SoundLoader
 from kivy.factory import Factory
+from kivy.uix.accordion import Accordion
+
 import meditation
 import relaxation
 import yoga
@@ -15,11 +16,32 @@
 
 class PracticeTimerApp(App):
 
+    _autoconf_widgets = {}
+
+    def autoconf(self, widget, module, key, default):
+        module_config = self._autoconf_widgets.setdefault(module, {})
+        module_config[key] = widget
+        self.config.setdefaults(module, {key: default})
+        return float(self.config.get(module, key))
+
+    def on_stop(self):
+        for module,keylist in self._autoconf_widgets.items():
+            for key,widget in keylist.items():
+                self.config.set(module, key, widget.value)
+                #print ">>>",module, key, widget.value
+        self.config.write()
+
+    def on_start(self):
+        #page = self.config.get("root", "selected")
+        #print page
+        for w in self.root.children:
+            print ">>>", w.title, w.collapse
+            if w.title=='High Intensity Interval Training':
+                w.collapse = False
+
     def build_config(self, config):
-        hiit.setdefaults(config)
-        meditation.setdefaults(config)
-        relaxation.setdefaults(config)
-        yoga.setdefaults(config)
+        # the config need to be built not empty to be saved
+        config.setdefaults('root', {})
 
 if __name__ == '__main__':
     Factory.register('HiitTimer', cls=hiit.HiitTimer)
diff -r ac61bf8c09a6 meditation.py
--- a/meditation.py	Fri Oct 05 17:50:58 2012 -0700
+++ b/meditation.py	Sun Oct 07 16:21:26 2012 -0700
@@ -1,14 +1,7 @@
-import kivy
-kivy.require('1.4.1')
-
 from kivy.clock import Clock
 from kivy.properties import ObjectProperty
-from kivy.uix.boxlayout import BoxLayout
-from kivy.uix.button import Button
-from kivy.uix.label import Label
-from kivy.uix.popup import Popup
 
-from time import GenTimer
+from timer import GenTimer
 from sequence import Sequence
 
 class MeditationTimer(GenTimer):
@@ -22,8 +15,12 @@
 
     def __init__(self, **kwargs):
         super(MeditationTimer, self).__init__(**kwargs)
-        self.sequence_begin = Sequence([clap] * 5 + [self.run_timer])
-        self.sequence_end   = Sequence([clap] * 5 + [gong])
+
+        #self.duration_init  = int(self.get_config('duration'))
+        #self.interval_init  = int(self.get_config('interval'))
+
+        self.sequence_begin = Sequence([self.clap] * 5 + [self.run_timer], 2.)
+        self.sequence_end   = Sequence([self.clap] * 5 + [self.gong], 2.)
 
     def increment_value(self, dt):
         self.progress.value += dt / self.total_duration
@@ -33,10 +30,10 @@
 
     def startstop(self, auto=False):
         if self.button.state == 'down':
-            set_config({
-                'duration': int(self.duration.value)
-                'interval': int(self.interval.value)
-            })
+            #self.set_config({
+            #    'duration': int(self.duration.value),
+            #    'interval': int(self.interval.value)
+            #})
             self.progress.value = 0
             self.total_duration = self.duration.value * 60. / 100.
             self.sequence_begin.start()
@@ -53,15 +50,10 @@
             Clock.schedule_interval(self.clap, self.interval.value * 60)
         self.gong()
 
-def setdefaults(config):
-    config.setdefaults('meditation', {
-        'duration': '5',
-        'interval': '1'
-    })
+    def help(self):
+        GenTimer.help('How to Meditate', help_text)
 
-help_title = 'How to Meditate'
-
-help_text = """Practice meditation once a week or more to understand how your body and mind work. Your goal is to observe yourself, as an external observer would, just like you would observe a feather gently floating in the air, making sure you are not disturbing it.
+help_text = """Meditation once a week or more to better understand how your body and mind work. Your goal is to observe yourself, as an external observer would, just like you would observe a feather gently floating in the air, making sure you are not disturbing it.
 
 Find a quiet place to sit comfortably, without sensory distractions. Start the timer and use the first 10 seconds to relax your body. Your arms should be resting on your legs or on the armrests of your chair. Breath normally, do not move your body, and keep your mind quiet throughout the exercise: do not use your mental voice to express your thoughts. You can close your eyes or keep them slightly open, but do not fall asleep.
 
diff -r ac61bf8c09a6 practicetimer.kv
--- a/practicetimer.kv	Fri Oct 05 17:50:58 2012 -0700
+++ b/practicetimer.kv	Sun Oct 07 16:21:26 2012 -0700
@@ -32,7 +32,7 @@
                 size_hint_y: .1
             Slider:
                 id: med_duration
-                value: int(med.get_config('duration'))
+                value: app.autoconf(self,"meditation", "duration", 15)
                 range: (0,150)
                 step: 5
                 size_hint_y: .2
@@ -40,12 +40,13 @@
             Label:
                 size_hint_y: .1
             Label:
-                text: 'Tick every %s minutes' % int(med_interval.value) if med_interval.value else 'No ticks'
+                text: 'Tick every %s %s' % (int(med_interval.value), med.plural(med_interval.value, 'minute', 'minutes')) if med_interval.value else 'No ticks'
                 size_hint_y: .1
             Slider:
                 size_hint_y: .2
                 id: med_interval
-                value: int(med.get_config('interval'))
+                value: app.autoconf(self,"meditation", "interval", 3)
+                #med.interval_init
                 range: (0,20)
                 step: 1
 
@@ -102,7 +103,7 @@
                 size_hint_y: .1
             Slider:
                 id: relx_duration
-                value: int(relx.get_config('duration'))
+                value: app.autoconf(self,"relaxation", "duration", 3)
                 range: (1,10)
                 step: 1
                 size_hint_y: .2
@@ -115,25 +116,25 @@
             Slider:
                 size_hint_y: .2
                 id: relx_interval
-                value: int(relx.get_config('interval'))
+                value: app.autoconf(self,"relaxation", "interval", 10)
                 range: (5,20)
-                step: 1
+                step: .5
 
             Label:
                 size_hint_y: None
             ToggleButton:
                 id: relx_button
-                font_size: 18
+                font_size: 14
                 halign: 'center'
                 markup: True
-                text: ('Press to Begin' if self.state == 'normal' else 'running ...')+'\n[size=32][b]%s[/b][/size]\n%s%%' % (relx.time(rel_progress.value), int(round(rel_progress.value)))
+                text: ('Press to Begin' if self.state == 'normal' else 'running ...')+'\n[size=32][b]%s[/b][/size]\n%s%%' % (relx.time(relx_progress.value), int(round(relx_progress.value)))
                 on_press: relx.startstop()
                 size_hint_x: .67
                 pos_hint: {'center_x': 0.5}
                 color: (0., 0., 0., 1.)
                 background_color: (1.0, 0.8, 0.2, 1.0)
             ProgressBar:
-                id: rel_progress
+                id: relx_progress
                 size_hint_y: .2
 
 
@@ -143,7 +144,7 @@
         YogaTimer:
             id: yoga
             button:   yoga_button
-            duration: yoga_duration
+            numpose:  yoga_numpose
             interval: yoga_interval
             progress: yoga_progress
 
@@ -158,12 +159,19 @@
                 pos_hint: {'right': 1.0}
                 on_release: yoga.help()
 
+            #Label:
+            #    text: 'Total duration: %s' % yoga.timetotal(yoga_numpose.value*(yoga_interval.value*60+5))
+            #    font_size: 20
+            #    size_hint_y: .1
+            #Label:
+            #    size_hint_y: .2
+
             Label:
                 text: 'Yoga session of %s poses' % int(yoga_numpose.value) if yoga_numpose.value else 'Unspecified number of poses'
                 size_hint_y: .1
             Slider:
                 id: yoga_numpose
-                value: int(yoga.get_config('duration'))
+                value: app.autoconf(self,"yoga", "num_poses", 10)
                 range: (0,30)
                 step: 1
                 size_hint_y: .2
@@ -176,7 +184,7 @@
             Slider:
                 size_hint_y: .2
                 id: yoga_interval
-                value: yoga.get_config('interval')
+                value: app.autoconf(self,"yoga", "interval", 5)
                 range: (1,10)
                 step: .5
 
@@ -187,7 +195,7 @@
                 font_size: 18
                 halign: 'center'
                 markup: True
-                text: ('Press to Begin' if self.state == 'normal' else 'running ...')+'\n[size=32][b]%s[/b][/size]\n[size=24](total %s)[/size]\n%s%%' % (yoga.time(yoga_progress.value), int(round(yoga_progress.value)), yoga.time(yoga_duration.value*yoga_numpose.value)[0])
+                text: ('Press to Begin' if self.state == 'normal' else 'running ...')+'\n[size=32][b]%s[/b][/size]\n[size=24](total %s)[/size]\n%s%%' % (yoga.time(yoga_progress.value), yoga.timetotal(yoga_numpose.value*(yoga_interval.value*60+5)), int(round(yoga_progress.value)))
                 on_press: yoga.startstop()
                 size_hint_x: .67
                 pos_hint: {'center_x': 0.5}
@@ -224,7 +232,7 @@
                 size_hint_y: .1
             Slider:
                 id: hiit_duration
-                value: hiit.get_config('duration')
+                value: app.autoconf(self,"hiit", "num_cycles", 10)
                 range: (1,10)
                 step: 1
                 size_hint_y: .2
@@ -237,7 +245,8 @@
             Slider:
                 size_hint_y: .2
                 id: hiit_interval
-                value: hiit.get_config('interval')
+                value: app.autoconf(self, "hiit", "high_intensity", 60)
+                #value: app.autoconf(self, "hiit", "low_intensity", 90)
                 range: (5,20)
                 step: 1
 
diff -r ac61bf8c09a6 relaxation.py
--- a/relaxation.py	Fri Oct 05 17:50:58 2012 -0700
+++ b/relaxation.py	Sun Oct 07 16:21:26 2012 -0700
@@ -1,15 +1,10 @@
-import kivy
-kivy.require('1.4.1')
+from kivy.clock import Clock
+from kivy.properties import ObjectProperty
 
-from kivy.app import App
-from kivy.clock import Clock
-from kivy.config import Config
-from kivy.core.audio import SoundLoader
-from kivy.properties import ObjectProperty
-from kivy.uix.boxlayout import BoxLayout
-from kivy.uix.button import Button
-from kivy.uix.label import Label
-from kivy.uix.popup import Popup
+from time import sleep
+
+from timer import GenTimer
+from sequence import Sequence
 
 class RelaxationTimer(GenTimer):
 
@@ -18,20 +13,13 @@
     interval = ObjectProperty()
     progress = ObjectProperty()
 
-    total_duration = 0
     config_name = 'relaxation'
 
     def __init__(self, **kwargs):
         super(RelaxationTimer, self).__init__(**kwargs)
-        self.tick_sound = SoundLoader.load('claves.ogg')
-        self.end_sound = SoundLoader.load('gong2.ogg')
-        self.sequence_begin = [self.tick] * 5 + [self.run_timer, self.stop_sequence]
-        self.sequence_end   = [self.tick] * 5 + [self.gong, self.stop_sequence]
 
-    def get_config(self, key):
-        val = App.get_running_app().config.get('relaxation', key)
-        print "(CONFIG ) (relaxation   )", key, "=", val
-        return int(val)
+        #self.duration_init  = int(self.get_config('duration'))
+        #self.interval_init  = int(self.get_config('interval'))
 
     def increment_value(self, dt):
         self.progress.value += dt / self.total_duration
@@ -39,57 +27,44 @@
           self.button.state = 'normal'
           self.startstop(True)
 
-    def time(self, t):
-        tsec = int(round(t * self.total_duration))
-        tmin = tsec / 60
-        tsec = tsec % 60
-        return ("%s min %s sec" % (tmin, tsec), int(round(t)))
-
     def startstop(self, auto=False):
         if self.button.state == 'down':
-            config = App.get_running_app().config
-            config.set('relaxation', 'duration', int(self.duration.value))
-            config.set('relaxation', 'interval', int(self.interval.value))
-            config.write()
+            #self.set_config({
+            #    'duration': int(self.duration.value),
+            #    'interval': int(self.interval.value)
+            #})
             self.progress.value = 0
             self.total_duration = self.duration.value * 60. / 100.
-            self.start_sequence(self.sequence_begin)
+            mult = 10
+            dur = int(round(self.interval.value*1.2 - .5*13))
+            if dur<3:
+                mult = 5
+                dur = int(round(self.interval.value*1.5 - .5*8))
+            print "*** dur=",dur
+            self.sequence_end = Sequence([self.clap, self.do_nothing, self.do_nothing, self.clap, self.do_nothing] +
+                                         [self.clap]*mult + [self.do_nothing]*dur + [self.gong], .5)
+            self.run_timer()
         else:
             Clock.unschedule(self.increment_value)
-            Clock.unschedule(self.tick)
-            self.stop_sequence()
+            Clock.unschedule(self.toggle_clap)
             if auto:
-                self.start_sequence(self.sequence_end)
+                self.sequence_end.start(debug=True)
 
-    def start_sequence(self, l):
-        self.sequence_list = l[:]
-        Clock.schedule_interval(self.run_sequence, 2.0)
-        self.run_sequence()
+    def toggle_clap(self, dt=0):
+        self.clap()
+        if self.toggle: # double clap
+            sleep(.15)
+            self.clap()
+        self.toggle = not self.toggle
 
-    def stop_sequence(self, dt=0):
-        Clock.unschedule(self.run_sequence)
-
-    def tick(self, dt=0):
-        self.tick_sound.play()
-
-    def gong(self, dt=0):
-        self.end_sound.play()
-
-    def run_sequence(self, dt=0):
-        #print self.sequence_list
-        self.sequence_list.pop(0)(dt)
-
-    def run_timer(self, dt=0):
+    def run_timer(self):
         Clock.schedule_interval(self.increment_value, .1)
-        if self.interval.value:
-            Clock.schedule_interval(self.tick, self.interval.value * 60)
+        self.toggle = False
+        Clock.schedule_interval(self.toggle_clap, self.interval.value)
         self.gong()
 
-def setdefaults(config):
-    config.setdefaults('relaxation', {
-        'duration': '5',
-        'interval': '10'
-    })
+    def help(self):
+        GenTimer.help(help_title, help_text)
 
 help_title = 'Breathing Exercise for Relaxation'
 
@@ -97,7 +72,7 @@
 
 Find a place to sit, or you can just stand. Keep your eyes slightly open and choose a spot in front to look at for the duration of the exercise.
 
-Start the timer and at the first clap, breathing in with your nose, slowly and deeply. At the double clap, breath out by your mouth, slowly and deeply. Keep breathing in with your nose at the single claps, and breathing out with your mouth at the double claps. Concentrate only on following the breathing rythm. (If you're out of breath and need to breath faster at the begining, just do the breathing out until you can follow the rythm.)
+Start the timer, wait after the gong, and at the first clap, breath in with your nose, slowly and deeply. At the double clap, breath out by your mouth, slowly and deeply. Keep breathing in with your nose at the single claps, and breathing out with your mouth at the double claps. Concentrate only on following the breathing rythm. (If you're out of breath and need to breath faster at the begining, just do the breathing out until you can follow the rythm.)
 
 At the end of the exercise, you will hear multiple claps for the last breath in. This is to tell you that the last one will be slightly longer, so hold it in. At the gong sound, breath out: the exercise is finished. Your body should be relaxed and your mind clear.
 """
diff -r ac61bf8c09a6 sequence.py
--- a/sequence.py	Fri Oct 05 17:50:58 2012 -0700
+++ b/sequence.py	Sun Oct 07 16:21:26 2012 -0700
@@ -1,6 +1,3 @@
-import kivy
-kivy.require('1.4.1')
-
 from kivy.clock import Clock
 
 class Sequence:
@@ -9,14 +6,20 @@
         self.l = l
         self.interval = i
 
-    def start(self):
+    def start(self, debug=False):
         self.todo = self.l[:]
+        self.debug = debug
         Clock.schedule_interval(self.run_item, self.interval)
-        self.run_sequence()
+        self.run_item(0)
 
     def stop(self):
-        Clock.unschedule(self.run_sequence)
+        Clock.unschedule(self.run_item)
 
     def run_item(self, dt):
-        self.todo.pop(0)(dt)
+        if self.debug:
+            print "(DEBUG  ) (Sequence    )", [ x.__name__ for x in self.todo]
+        if self.todo:
+            self.todo.pop(0)(dt)
+        else:
+          self.stop()
 
diff -r ac61bf8c09a6 timer.py
--- a/timer.py	Fri Oct 05 17:50:58 2012 -0700
+++ b/timer.py	Sun Oct 07 16:21:26 2012 -0700
@@ -1,6 +1,3 @@
-import kivy
-kivy.require('1.4.1')
-
 from kivy.app import App
 from kivy.clock import Clock
 from kivy.core.audio import SoundLoader
@@ -9,36 +6,34 @@
 from kivy.uix.label import Label
 from kivy.uix.popup import Popup
 
+import env
+
 class GenTimer(BoxLayout):
 
-    clap_sound = SoundLoader.load('claves.ogg')
-    gong_sound = SoundLoader.load('gong2.ogg')
-
     config_name = "all_timer"
-    config = App.get_running_app().config
 
     def __init__(self, **kwargs):
         super(GenTimer, self).__init__(**kwargs)
         self.total_duration = 0
+        env.init()
 
-    @classmethod
-    def get_config(key):
-        val = config.get(config_name, key)
-        print "(CONFIG ) (%12s)" % (config_name, key, val)
-        return val
+    #def get_config(self, key):
+    #    val = App.get_running_app().config.get(self.config_name, key)
+    #    print "(CONFIG ) (%-12s) %s = %s" % (self.config_name, key, val)
+    #    return val
 
-    @classmethod
-    def set_config(m):
-        for k,v in m:
-            config.set(config_name, k, v)
+    def set_config(self, m):
+        app = App.get_running_app()
+        for k,v in m.items():
+            app.config.set(self.config_name, k, v)
+        for w in app.root.children:
+            if not w.collapse:
+                app.config.set("root", "selected", w.title)
+                #print ">>>",w.title, w.collapse
+        app.config.write()
 
-    @classmethod
-    def set_defaults(m):
-        config.setdefaults(config_name, m)
-
-    @classmethod
-    def write_config():
-        config.write()
+    def set_defaults(self, m):
+        App.get_running_app().config.setdefaults(self.config_name, m)
 
     def time(self, t):
         tsec = int(round(t * self.total_duration))
@@ -46,36 +41,35 @@
         tsec = tsec % 60
         return "%s min %s sec" % (tmin, tsec)
 
+    def timetotal(self, t):
+        tsec = int(round(t))
+        tmin = tsec / 60
+        tsec = tsec % 60
+        return "%s min %s sec" % (tmin, tsec)
+
+    @staticmethod
+    def clap(dt=0):
+        env.clap_sound.stop()
+        env.clap_sound.play()
+
+    @staticmethod
+    def gong(dt=0):
+        env.gong_sound.stop()
+        env.gong_sound.play()
+
+    @staticmethod
+    def do_nothing(dt=0):
+        pass
+
     @staticmethod
     def plural(num, sing, plur):
-        if 0<num && num<=1:
+        if 0<num and num<=1:
             return sing
         else:
             return plur
 
-    @classmethod
-    def clap(dt=0):
-        clap_sound.stop()
-        clap_sound.play()
-
-    @classmethod
-    def gong(dt=0):
-        gong_sound.stop()
-        gong_sound.play()
-
-    def start_sequence(self, l):
-        self.sequence_list = l[:]
-        Clock.schedule_interval(self.run_sequence, 2.0)
-        self.run_sequence()
-
-    def stop_sequence(self, dt=0):
-        Clock.unschedule(self.run_sequence)
-
-    def run_sequence(self, dt=0):
-        #print self.sequence_list
-        self.sequence_list.pop(0)(dt)
-
-    def help(self):
+    @staticmethod
+    def help(help_title, help_text):
         btnclose = Button(text='close', size_hint_y=None, height=50)
         content = BoxLayout(orientation='vertical')
         content.add_widget(Label(text=help_text, halign='left', valign='middle', text_size=(420,680)))
diff -r ac61bf8c09a6 yoga.py
--- a/yoga.py	Fri Oct 05 17:50:58 2012 -0700
+++ b/yoga.py	Sun Oct 07 16:21:26 2012 -0700
@@ -1,15 +1,8 @@
-import kivy
-kivy.require('1.4.1')
+from kivy.clock import Clock
+from kivy.properties import ObjectProperty
 
-from kivy.app import App
-from kivy.clock import Clock
-from kivy.config import Config
-from kivy.core.audio import SoundLoader
-from kivy.properties import ObjectProperty
-from kivy.uix.boxlayout import BoxLayout
-from kivy.uix.button import Button
-from kivy.uix.label import Label
-from kivy.uix.popup import Popup
+from timer import GenTimer
+from sequence import Sequence
 
 class YogaTimer(GenTimer):
 
@@ -20,17 +13,12 @@
 
     total_duration = 0
 
+    config_name = 'yoga'
+
     def __init__(self, **kwargs):
         super(YogaTimer, self).__init__(**kwargs)
-        self.tick_sound = SoundLoader.load('claves.ogg')
-        self.end_sound = SoundLoader.load('gong2.ogg')
-        self.sequence_begin = [self.tick] * 5 + [self.run_timer, self.stop_sequence]
-        self.sequence_end   = [self.tick] * 5 + [self.gong, self.stop_sequence]
-
-    def get_config(self, key):
-        val = App.get_running_app().config.get('yoga', key)
-        print "(CONFIG ) (yoga         )", key, "=", val
-        return int(val)
+        #self.sequence_begin = [self.clap] * 5 + [self.run_timer, #self.stop_sequence]
+        #self.sequence_end   = [self.clap] * 5 + [self.gong, self.stop_sequence]
 
     def increment_value(self, dt):
         self.progress.value += dt / self.total_duration
@@ -38,12 +26,6 @@
           self.button.state = 'normal'
           self.startstop(True)
 
-    def time(self, t):
-        tsec = int(round(t * self.total_duration))
-        tmin = tsec / 60
-        tsec = tsec % 60
-        return ("%s min %s sec" % (tmin, tsec), int(round(t)))
-
     def startstop(self, auto=False):
         if self.button.state == 'down':
             config = App.get_running_app().config
@@ -55,59 +37,31 @@
             self.start_sequence(self.sequence_begin)
         else:
             Clock.unschedule(self.increment_value)
-            Clock.unschedule(self.tick)
+            Clock.unschedule(self.clap)
             self.stop_sequence()
             if auto:
                 self.start_sequence(self.sequence_end)
 
-    def start_sequence(self, l):
-        self.sequence_list = l[:]
-        Clock.schedule_interval(self.run_sequence, 2.0)
-        self.run_sequence()
-
-    def stop_sequence(self, dt=0):
-        Clock.unschedule(self.run_sequence)
-
-    def tick(self, dt=0):
-        self.tick_sound.play()
-
-    def gong(self, dt=0):
-        self.end_sound.play()
-
-    def run_sequence(self, dt=0):
-        #print self.sequence_list
-        self.sequence_list.pop(0)(dt)
-
     def run_timer(self, dt=0):
         Clock.schedule_interval(self.increment_value, .1)
         if self.interval.value:
-            Clock.schedule_interval(self.tick, self.interval.value * 60)
+            Clock.schedule_interval(self.clap, self.interval.value * 60)
         self.gong()
 
     def help(self):
-        btnclose = Button(text='close', size_hint_y=None, height=50)
-        content = BoxLayout(orientation='vertical')
-        content.add_widget(Label(text=help_text, halign='left', valign='middle', text_size=(420,640)))
-        content.add_widget(btnclose)
-        popup = Popup(content=content, title='How to Practice Yoga',
-                      auto_dismiss=False)
-        btnclose.bind(on_release=popup.dismiss)
-        popup.open()
-
-def setdefaults(config):
-    config.setdefaults('yoga', {
-        'num_poses': '15',
-        'interval': '5'
-    })
+        GenTimer.help(help_title, help_text)
 
 help_title = 'How to Practice Yoga'
 
 help_text = """Practice yoga once a week or more to develop your flexibility, strength, and balance.
 
+
 Wear comfortable clothing and warm up before every session. Go slow, know your limits and stay within them.
 
+
 Start the timer and prepare for the first pose. Keep the pose until the gong sound. You will have a few seconds marked by claps to get into the next pose.
 
+
 The session end with a triple gong sound. Take a few deep breaths, slowly stretch your arms, your legs and your back.
 """
 
